apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chat/ComposeView.kt

```diff
@@ -429,16 +429,21 @@ fun ComposeView(
           val tmpFile = File(preview.voice)
           AudioPlayer.stop(tmpFile.absolutePath)
           val actualFile = File(getAppFilePath(tmpFile.name.replaceAfter(RecorderInterface.extension, "")))
-          withContext(Dispatchers.IO) {
-            Files.move(tmpFile.toPath(), actualFile.toPath())
-          }
-          // TODO encrypt voice files
-          files.add(CryptoFile.plain(actualFile.name))
+          files.add(withContext(Dispatchers.IO) {
+            if (chatController.appPrefs.privacyEncryptLocalFiles.get()) {
+              val args = encryptCryptoFile(tmpFile.absolutePath, actualFile.absolutePath)
+              tmpFile.delete()
+              CryptoFile(actualFile.name, args)
+            } else {
+              Files.move(tmpFile.toPath(), actualFile.toPath())
+              CryptoFile.plain(actualFile.name)
+            }
+          })
           deleteUnusedFiles()
           msgs.add(MsgContent.MCVoice(if (msgs.isEmpty()) msgText else "", preview.durationMs / 1000))
         }
-        is ComposePreview.FilePreview -> {
-          val file = saveFileFromUri(preview.uri, encrypted = false)
+        is ComposePreview.FilePreview -> { // LALAL Not sure we need to save encrypted because it will be deleted anyway
```

Comment:
in iOS we do it because file access is session scoped, and we need resumable upload in the next session. Also app upgrades change paths it seems.

It's the same question as for unencrypted file - if the path and access pesists between sessions, maybe we don't need to copy it at all... But Haskell code assumes the file is in app folder, so it'll be not a small change to support files in other locations...